accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+# Improve support for NuGet packages that are signed with a Trusted Signing certificate
+
+- Author Name: Dirk Lemstra [dlemstra](https://github.com/dlemstra)
+- GitHub Issue: https://github.com/NuGet/NuGetGallery/issues/10027
+
+## Summary
+
+This proposal is about improving the support for NuGet packages that are signed with a Trusted Signing certificate. This proposal
+consists of three steps to improve the experience for users that sign their packages with Trusted Signing. These three steps can
+be released separately but I can also understand that the NuGet team would want to launch this as a complete feature.
+
+## Motivation
+
+Earlier this year [Trusted Signing](https://learn.microsoft.com/en-us/azure/trusted-signing/) was launched by Microsoft and 
+recently support for this was added to `dotnet sign`. Packages signed with this can be uploaded to the NuGet Gallery. But
+because the certificate is only valid for three days a user will probably always need to update their certificate on their 
+account page. I would like to propose a set of changes to improve the experience for users that sign their NuGet package
+with Trusted Signing.
+
+## Explanation
+
+### Functional explanation
+
+### Step 1: Register EKU of certificate that was added by Trusted Signing
```

Comment:
I think we can implement each step individually. I am concerned that Step 1 has a part this is not self-service (the deletion, potentially increasing on-call/support load) and Step 2 requires the user to extract the .cer themselves somehow. This is fine, but we can limit exposure but putting the new behavior behind a flight -- this is a feature flag but has a username filter. This means we can add specific people (perhaps MVPs who ask) to the feature until we get to step 3 and then we can open it up to everyone.

What do you think?

Reply:
Sounds like a good plan. Not sure if we should limit this to just MVP's though. There are probably also other people that would want to try out this feature.

Reply:
Yes, I take MVPs as an example because it's easy to have tight, proactive communication with them. But certainly, we can definitely open it up to anyone that asks. Maybe a small announcement with "contact us if you want to try it", or perhaps a set of people Ian knows. Initial adoption efforts can be TBD. We can add any username to the flight system with ease.